pkg/kubectl/cmd/get.go

```diff
@@ -168,7 +180,8 @@ func RunGet(f *cmdutil.Factory, out io.Writer, errOut io.Writer, cmd *cobra.Comm
 
 	if len(args) == 0 && cmdutil.IsFilenameEmpty(options.Filenames) {
 		fmt.Fprint(errOut, "You must specify the type of resource to get. ", valid_resources)
-		return cmdutil.UsageError(cmd, "Required resource not specified.")
+		usageString := fmt.Sprintf("Required resource not specified.\nUse \"%s explain <resource>\" for a detailed description of that resource (e.g. %[1]s explain pods).", options.FullCmdName)
```

Comment:
You don't need to pass `FullCmdName` in. You can use the passed-in `cmd` and do the `cmd. cmds.Root().CommandPath()` here.